api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -102,6 +102,38 @@ public static <T> UnboundTerm<T> truncate(String name, int width) {
     return new UnboundTransform<>(ref(name), Transforms.truncate(width));
   }
 
+  @SuppressWarnings("unchecked")
+  public static <S, T> UnboundTerm<T> bucket(ResolvedReference<S> resolvedRef, int numBuckets) {
```

Comment:
There are already transform functions that accept `UnboundTerm`. Why are these needed?

Reply:
```
  public static <T> UnboundTerm<T> bucket(String name, int numBuckets) {
    Transform<?, T> transform = (Transform<?, T>) Transforms.bucket(numBuckets);
    return new UnboundTransform<>(ref(name), transform);
  }
```

The signatures for those are tied to the String name, i can move this the UnboundTerm as even in this function calls `ref(name)` which returns NamedReference 

this way we have just one api and the caller needs to make sure it passes appropriate unboundTerm